src/reactions/MessageReactionList.js

```diff
@@ -0,0 +1,116 @@
+
+import ReactionUserList from './ReactionUserList';
+import { connect } from '../react-redux';
+import type { AggregatedReaction, Dispatch, Message, UserOrBot } from '../types';
+import { Screen, Label, RawLabel } from '../common';
+import { getOwnUser } from '../selectors';
+import aggregateReactions from './aggregateReactions';
+import styles from '../styles';
+import { getAllUsersById } from '../users/userSelectors';
+import tabsOptions from '../styles/tabs';
+import Emoji from '../emoji/Emoji';
+import { objectFromEntries } from '../jsBackport';
+
+const getReactionsTabs = (
+  aggregatedReactions: $ReadOnlyArray<AggregatedReaction>,
+  reactionName: string,
+  usersById: Map<number, UserOrBot>,
+) => {
+  const reactionsTabs = objectFromEntries(
+    aggregatedReactions.map(aggregatedReaction => [
+      aggregatedReaction.name,
+      {
+        screen: () => (
+          <ReactionUserList reactedUserIds={aggregatedReaction.users} users={usersById} />
+        ),
+        navigationOptions: {
+          tabBarLabel: () => (
+            <View style={styles.row}>
+              <Emoji name={aggregatedReaction.name} />
+              <RawLabel style={styles.paddingLeft} text={`${aggregatedReaction.count}`} />
+            </View>
+          ),
+        },
+      },
+    ]),
+  );
+
+  return TabNavigator(reactionsTabs, {
+    backBehavior: 'none',
+    initialRouteName: reactionName,
+    tabBarComponent: TabBarTop,
+    tabBarPosition: 'top',
+    ...tabsOptions({
+      showLabel: true,
+      showIcon: false,
+      style: {
+        borderWidth: 0.15,
+      },
+    }),
+  });
+};
+
+type NavigationsProps = {|
+  navigation: NavigationScreenProp<{ params: {| reactionName: string, messageId: number |} }>,
+|};
+
+type SelectorProps = {|
+  message: Message,
+  ownUserId: number,
+  usersById: Map<number, UserOrBot>,
+|};
+
+type Props = {|
+  dispatch: Dispatch,
+  ...NavigationsProps,
+  ...SelectorProps,
+|};
+
+class MessageReactionList extends PureComponent<Props> {
+  render() {
+    const { message, navigation, ownUserId, usersById } = this.props;
+    const { reactionName } = navigation.state.params;
+    const { reactions } = message;
+
+    // useful when user is on this screen and reactions are revoked
+    if (reactions.length === 0) {
+      return (
+        <Screen title="Reactions" scrollEnabled={false}>
+          <View style={[styles.flexed, styles.center]}>
+            <Label style={styles.largerText} text="No reactions" />
+          </View>
+        </Screen>
+      );
+    }
+
+    const aggregatedReactions = aggregateReactions(reactions, ownUserId);
+
+    // there can be cases where reaction have been removed by the time user see this screen
+    // or on re-render
+    // so confirm reaction `reactionName` still exists
+    const TabView = getReactionsTabs(
+      aggregatedReactions,
+      aggregatedReactions.find(reaction => reaction.name === reactionName) ? reactionName : '',
```

Comment:
This would be a bit cleaner after `reactionsTabs` has been constructed, inside `getReactionsTabs`.

Reply:
Updated 👍